drivers/ethernet/Kconfig.native_posix

```diff
@@ -5,11 +5,11 @@
 # SPDX-License-Identifier: Apache-2.0
 
 menuconfig ETH_NATIVE_POSIX
-	bool
-	prompt "Native Posix Ethernet driver"
+	bool "Native Posix Ethernet driver"
 	depends on ARCH_POSIX
-	select NET_L2_ETHERNET
+	select NET_L2_ETHERNET if NETWORKING
 	default n
+	default y if NETWORKING && BOARD_NATIVE_POSIX
```

Comment:
just put CONFIG_ETH_NATIVE_POSIX=y in boards/posix/native_posix/native_posix_defconfig instead..

Reply:
Wouldn't that just break everything like in his 1st proposal?

Reply:
Yes, I think I proposed this in v1. But that would have introduced ethernet driver always, and we only want it if networking is enabled.